developers/contracts/cross-peg-out-endpoint-v2-01.md

```diff
@@ -0,0 +1,136 @@
+# cross-peg-out-endpoint-v2-01.clar
+
+- Location: `xlink/packages/contracts/bridge-stacks/contracts/cross-peg-out-endpoint-v2-01.clar`
+- [Deployed contract]()
+
+This technical document provides a detailed overview of the contract responsible for managing the peg-out process, enabling the transfer of `SIP-010` bridged tokens from the Stacks network to EVM-compatible blockchain networks as the EVM-based assets. The contract manages token transfers by validating amounts, applying fees, and maintaining a whitelist for authorized users if enabled. During the process, tokens are either burned or transferred (depending on the token's configuration) to a designated address on the EVM chain. The core functionalities of the contract are implemented through a series of public and governance functions, as described below.
+
+## Storage
+### `is-paused`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `bool` |
+
+A flag that indicates whether the peg-out process is active. If set to `true`, all peg-out operations are paused, preventing any new transactions. The contract is deployed in a paused state by default.
+
+### `use-whitelist`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `bool` |
+
+A flag that determines whether a whitelist is enforced for peg-out operations. If set to `true`, only addresses explicitly whitelisted can execute peg-out transactions. By default, this feature is disabled.
+
+### `whitelisted-users`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `principal` `bool` |
+
+This map stores the whitelist status of users. Each entry associates a `principal` with a boolean value indicating whether they are authorized to perform peg-out transactions. If `true`, the user is whitelisted; otherwise, they are not permitted to participate in the process.
```

Comment:
I think I would add: "**When `use-whitelist` if  this value is** `true`, the user is whitelisted; other..."  or something else..

As it is now, it seems that this applies always.. a comment about the `use-whitelist` to enable tihs is required here.